src/std/misc/list.ss

```diff
@@ -449,3 +450,18 @@
       ([a . r] (lp r (cons a positionals) keywords))
       ([] (values (reverse positionals) (reverse keywords))))))
 
+;; every-of returns true if all predicates match. If preds contains a
+;; non-predicate, it is transformed into one using equal? as test.
+;;
+;; Example:
+;;  (every-of 2 number? fixnum?) => #t
+(def (every-of v . preds)
+  (every (cut <> v) (map (lambda (p) (if (procedure? p) p (is p))) preds)))
```

Comment:
Let's put the procedure test and is transformation in the head predicate, so that we avoid the intermediate list.
Also, why not use `andmap` here? It's more efficient than `every`.